internal/core/queries/get_dd_by_ids.go

```diff
@@ -19,25 +22,33 @@ func (*GetDeviceDefinitionByIdsQuery) Key() string { return "GetDeviceDefinition
 
 type GetDeviceDefinitionByIdsQueryHandler struct {
 	DDCache services.DeviceDefinitionCacheService
+	log     *zerolog.Logger
 }
 
-func NewGetDeviceDefinitionByIdsQueryHandler(cache services.DeviceDefinitionCacheService) GetDeviceDefinitionByIdsQueryHandler {
+func NewGetDeviceDefinitionByIdsQueryHandler(cache services.DeviceDefinitionCacheService, log *zerolog.Logger) GetDeviceDefinitionByIdsQueryHandler {
 	return GetDeviceDefinitionByIdsQueryHandler{
 		DDCache: cache,
+		log:     log,
 	}
 }
 
 func (ch GetDeviceDefinitionByIdsQueryHandler) Handle(ctx context.Context, query mediator.Message) (interface{}, error) {
 
 	qry := query.(*GetDeviceDefinitionByIdsQuery)
 
+	if len(qry.DeviceDefinitionID) == 0 {
+		return nil, &exceptions.ValidationError{
+			Err: errors.New("Device Definition Ids is required"),
+		}
+	}
+
 	response := &grpc.GetDeviceDefinitionResponse{}
 
 	for _, v := range qry.DeviceDefinitionID {
 		dd, _ := ch.DDCache.GetDeviceDefinitionByID(ctx, v)
 
 		if dd == nil {
-			fmt.Printf("Not found")
+			ch.log.Warn().Msg(fmt.Sprintf("Device Definition Id %s not found", v))
```

Comment:
I liked the log message starting with "Not found" because made it easy to see in logs. Consider changing to:
`ch.log.Warn().Msg(fmt.Sprintf("Not found - deviceDefinitionId %s", v))`

Reply:
I like putting the variable stuff in the properties too.